pkg/graveler/committed/apply.go

```diff
@@ -26,99 +26,129 @@ func incrementDiffSummary(d *graveler.DiffSummary, typ graveler.DiffType) {
 
 // ReferenceType represents the type of the reference
 
-// applyFromSource applies all changes from source to writer.
-func applyFromSource(ctx context.Context, logger logging.Logger, writer MetaRangeWriter, source Iterator) error {
+// applyFromIter applies all changes from Iterator to writer.
+// returns the number of writes
+func applyFromIter(ctx context.Context, logger logging.Logger, writer MetaRangeWriter, iter Iterator) (int, error) {
+	var count int
 	for {
 		select {
 		case <-ctx.Done():
-			return ctx.Err()
+			return 0, ctx.Err()
 		default:
 		}
-
-		sourceValue, sourceRange := source.Value()
-		if sourceValue == nil {
-			if logger.IsTracing() {
-				logger.WithFields(logging.Fields{
-					"from": string(sourceRange.MinKey),
-					"to":   string(sourceRange.MaxKey),
-					"ID":   sourceRange.ID,
-				}).Trace("copy entire source range at end")
-			}
-			if err := writer.WriteRange(*sourceRange); err != nil {
-				return fmt.Errorf("copy source range %s: %w", sourceRange.ID, err)
+		iterValue, iterRange := iter.Value()
+		if iterValue == nil {
+			if iterRange.IsTombstone() {
+				// internal error but no data lost: deletion requested of a
+				// file that was not there.
+				logger.WithField("id", string(iterRange.ID)).Warn("[I] unmatched delete")
```

Comment:
This is a range, let's state as much!

```suggestion
				logger.WithFields(logging.Fields{
					"from": string(sourceRange.MinKey),
					"to":   string(sourceRange.MaxKey),
					"ID", string(iterRange.ID),
				}).Warn("[I] unmatched delete range")
```
(although, TBH, not sure why we don't use lowercase throughout for fields).